controllers/chat/ChatController.js

```diff
@@ -160,39 +186,140 @@ module.exports = {
+          const filter = { channelId: req.params.channelId };
+          if (dt.role === 'owner') {
+            filter.myAnonUserId = dt.user_id;
+          } else {
+            filter.targetUserId = dt.user_id;
+          }
+          const anonInfo = await ChatAnonUserInfo.findOne({ where: filter });
+          return {
+            ...dt,
+            anon_user_info_color_code: anonInfo.anon_user_info_color_code,
+            anon_user_info_color_name: anonInfo.anon_user_info_color_name,
+            anon_user_info_emoji_code: anonInfo.anon_user_info_emoji_code,
+            anon_user_info_emoji_name: anonInfo.anon_user_info_emoji_name,
+          };
+        } else {
+          return dt;
+        }
+      })
+    );
+
+    res.status(200).json(responseSuccess('Success', data));
   },
   sendMessage: async (req, res) => {
+    try {
+      const schema = {
+        members: 'string[]|empty:false',
+        message: 'string|empty:false',
+        anon_user_info_color_code: 'string|empty:false',
+        anon_user_info_color_name: 'string|empty:false',
+        anon_user_info_emoji_code: 'string|empty:false',
+        anon_user_info_emoji_name: 'string|empty:false',
+      };
+      const validated = v.validate(req.body, schema);
+      if (validated.length)
+        return res.status(403).json({
+          message: 'Error validation',
+          error: validated,
+        });
+      const {
+        members,
+        message,
+        anon_user_info_color_code,
+        anon_user_info_color_name,
+        anon_user_info_emoji_code,
+        anon_user_info_emoji_name,
+      } = req.body;
+      if (!members.includes(req.userId)) members.push(req.userId);
+
+      const client = StreamChat.getInstance(
+        process.env.API_KEY,
+        process.env.SECRET
+      );
+
+      await client.connectUser({ id: req.userId }, req.token);
+
+      const channel = client.channel('messaging', { members });
+
+      await channel.create();
+
+      const chat = await channel.sendMessage({
+        user_id: req.userId,
+        text: message,
+        ...req.body,
+      });
+
+      const targets = members.filter((member) => member !== req.userId);
+      targets.map(async (target) => {
+        await ChatAnonUserInfo.create({
+          channelId: channel.id,
+          myAnonUserId: req.userId,
+          targetUserId: target,
+          anon_user_info_color_code,
+          anon_user_info_color_name,
+          anon_user_info_emoji_code,
+          anon_user_info_emoji_name,
+        });
+      });
+      await saveChat({ user_id: req.userId, channelId: channel.id, message });
+
+      await client.disconnectUser();
+
+      return res.status(200).json(responseSuccess('sent', chat));
+    } catch (error) {
+      return ErrorResponse.e400(res, error.message);
+    }
+  },
+  getMyAnonProfile: async (req, res) => {
     const schema = {
-      members: 'string[]|empty:false',
-      message: 'string|empty:false',
+      channelId: 'string',
```

Comment:
Mas @falhar 
frontend can't get channel Id, so this validation should be removed.
Channel ID should be queried on the backend and frontend should not provide that at all.